packages/notarization/manager.go

```diff
@@ -469,12 +486,53 @@ func (m *Manager) outputsToOutputIDs(outputs devnetvm.Outputs) (createdIDs utxo.
 	return
 }
 
+// CheckIfEpochChanged check if next epoch started and trigger the event.
+func (m *Manager) CheckIfEpochChanged(issuingTime time.Time) {
+	currentTime := m.tangle.TimeManager.ATT()
+	if issuingTime.After(currentTime) {
+		currentTime = issuingTime
+	}
+	ei := epoch.IndexFromTime(currentTime)
+	currentEpochIndex, err := m.epochCommitmentFactory.storage.CurrentEpochIndex()
+	if err != nil {
+		m.log.Error(errors.Wrap(err, "could not get current epoch index"))
+		return
+	}
+	if ei > currentEpochIndex {
+		err = m.epochCommitmentFactory.storage.SetCurrentEpochIndex(ei)
+		if err != nil {
+			m.log.Error(errors.Wrap(err, "could not set current epoch index"))
+			return
+		}
+		if !m.isCommittable(ei) {
+			return
+		}
+		if err = m.epochCommitmentFactory.storage.SetLatestCommittableEpochIndex(ei); err != nil {
+			m.log.Error(errors.Wrap(err, "could not set latest committable epoch index"))
+			return
+		}
+		m.triggerManaVectorUpdate(ei)
+		// todo move commitments trees to the latest committable epoch
```

Comment:
Is this still a TODO?

Reply:
No, I forgot to remove it. At the beginning I had an idea to use this new epoch event to trigger update commitment roots, this way we will not do it on new message if commitment does not exist. I discarded it as what we have right now works fine